File: quotes/inject.py
import re

with open(r'c:\Users\mateo\Desktop\antigravy\website\quotes\Quotes Landing Page.html', 'r', encoding='utf-8') as f:
    html = f.read()

colors = ['orange', 'blue', 'emerald', 'purple', 'rose', 'grey']
idx = [0]

def inject_classes(match):
    color = colors[idx[0] % len(colors)]
    idx[0] += 1
    
    article_tag = match.group(0)
    if 'card-glow' not in article_tag:
        article_tag = article_tag.replace('masonry-card', f'masonry-card card-glow card-accent-{color}')
    return article_tag

html = re.sub(r'<article class="masonry-card([^"]*)"', inject_classes, html)

with open(r'c:\Users\mateo\Desktop\antigravy\website\quotes\Quotes Landing Page.html', 'w', encoding='utf-8') as f:
    f.write(html)

File: quotes/inject.js
const fs = require('fs');

const file = 'c:/Users/mateo/Desktop/antigravy/website/quotes/Quotes Landing Page.html';
let html = fs.readFileSync(file, 'utf8');

const colors = ['orange', 'blue', 'emerald', 'purple', 'rose', 'grey'];
let i = 0;

html = html.replace(/<article class="masonry-card([^"]*)"/g, (match, classes) => {
    if (match.includes('card-glow')) return match; // Already processed
    const color = colors[i % colors.length];
    i++;
    return `<article class="masonry-card card-glow card-accent-${color}${classes}"`;
});

fs.writeFileSync(file, html, 'utf8');
console.log('Done injecting classes.');
